frontend/simple/js/transactions.js

```diff
@@ -0,0 +1,263 @@
+'use strict'
+
+import localforage from 'localforage'
+import * as Events from '../../../shared/events'
+import sbp from '../../../shared/sbp'
+
+let _steps = new WeakMap()
+let _scope = new WeakMap()
+let _cursor = new WeakMap()
+let _description = new WeakMap()
+let _isRecoverable = new WeakMap()
+let _wait = new WeakMap()
```

Comment:
Finished reading `Transaction` class, and so far it's not clear to me at all why these `WeakMap`s are needed here and why internal instance properties aren't being used instead...